x-pack/plugins/observability_solution/dataset_quality/common/api_types.ts

```diff
@@ -37,6 +37,25 @@ export const dataStreamStatRt = rt.intersection([
 
 export type DataStreamStat = rt.TypeOf<typeof dataStreamStatRt>;
 
+export const dataStreamDocsStatRt = rt.type({
```

Comment:
This type is now generic, can be reused by `totalDocs` and later by `failedDocs`. There is no need to create one type per each